### PR TITLE
Fix 404 on file sharing.

### LINKED ISSUE #111: File sharing to other links leads to 404.
**Describe the bug**
When accessing a file through a shared link, and then pressing Assets etc it leads to a 404. Originally discovered by @cherry-john and myself.

Screenshot 2021-06-02 at 10 58 59

**Additional context**
The links are `https://inventory.yorktechsoc.org/file/assets.php` rather than `https://inventory.yorktechsoc.org/assets.php`. Should be a relatively easy fix. 

I would love to try and fix this issue but unsure where the code for inventory.techsoc.org is located.

### PR DESCRIPTION
**Pull request by chriscn**
Closes #111 

Before merging make sure that I have got the correct domain.

### CODE REVIEW

**Comment by chriscn** [comment]
After speaking with @Robert-Watts he believes that the first index is localhost and the second is the external domain. 

I’m not entirely sure so would want clarification.

**Comment by Robert-Watts** [comment]
I was just going through my messages with James

File: html/public/templatePublic.twig [hunk 1]
```diff
@@ -191,21 +191,21 @@
                         </li>
                         {% if INSTANCE.publicData.enableAssets %}
                             <li class="nav-item">
-                                <a class="nav-link" href="assets.php">
+                                <a class="nav-link" href="{{ INSTANCE.publicData.customDomains[0] }}/assets.php">
```
Comment (Robert-Watts) [inline html/public/templatePublic.twig:194]: I think currently you have `{{ INSTANCE.publicData.customDomains[0] }}/assets.php` but what if you get rid of the whole custom domain bit, and just have `/assets.php` is should fix the issue

Comment (Robert-Watts) [reply html/public/templatePublic.twig:194]: that way it does not matter what the domain is it is always correct

Comment (chriscn) [reply html/public/templatePublic.twig:194]: Whilst walking back, I realised that I could just use a relatively link.
Ergo `../assets.php` should work surely. But then it isn't as flexible and I would want an absolute link.

Comment (cherry-john) [inline html/public/templatePublic.twig:194]: Has this been tested with multiple custom domains? AFAIK this is always going to redirect to the first custom domain but the user could be accessing from a different domain which is also valid

**Comment by Jbithell** [comment]
@chriscn I've made you an outside collaborator to save you having to branch your own fork

**Review by cherry-john — CHANGES_REQUESTED** [review]
Not convinced this does the job at all? It doesn't allow for multiple custom domains

**Comment by Jbithell** [comment]
> Ergo ../assets.php should work surely. But then it isn't as flexible and I would want an absolute link.

@chriscn is there a particular reason you'd prefer absolute links?

**Comment by chriscn** [comment]
> > Ergo ../assets.php should work surely. But then it isn't as flexible and I would want an absolute link.
> 
> @chriscn is there a particular reason you'd prefer absolute links?

Yes, the file `templatePublic.twig` is used across multiple things (as a base template I believe). Therefor by setting it to use relatively links things are going to break more easily. I don't know what HTML does when it reaches the top of the root tree and you say to go up another directory. Does it just stay where it is?

File: html/public/templatePublic.twig (hunk 1 above)
Comment (chriscn) [reply html/public/templatePublic.twig:194]: In the event of multiple custom domains, does it matter if they get redirected to the first domain?

**Comment by Jbithell** [comment]
> > > Ergo ../assets.php should work surely. But then it isn't as flexible and I would want an absolute link.
> > 
> > 
> > @chriscn is there a particular reason you'd prefer absolute links?
> 
> Yes, the file `templatePublic.twig` is used across multiple things (as a base template I believe). Therefor by setting it to use relatively links things are going to break more easily. I don't know what HTML does when it reaches the top of the root tree and you say to go up another directory. Does it just stay where it is?

Ah sorry I think I might see the confusion. Twig does not parse links - they are plain text. `templatePublic.twig` is used by all pages on the "public sites" but the issue with your solution that @cherry-john has also identified is that it just picks the first domain defined in the list of custom domains for a business which would inadvertently leak this custom domain if it was not the one the user first accessed - and furthermore depending on internal DNS/firewall rules would actually be a blocked domain (say the first domain is `intranet-rms.company.com`, but this domain does not have a public DNS record).  

Prefacing all links with `/` is the most elegant solution given that the custom domains matcher (line 31 of `/html/public/common/head.php`) does not support anything other than root domains & subdomains by nature (ie paths are not supported such as `company.com/rms-public/`) so I'm afraid I don't really see the appeal of adding absolute links.

**Comment by Jbithell** [comment]
_Sorry about all those commits - just trying to fish the file that was merged in another PR out of it_

**Review by Jbithell — APPROVED** [review]

**Comment by Jbithell** [comment]
Thanks for your contribution @chriscn - sorry it turned out to be a little more complicated than I think we all thought at first but it's merged, tagged and building now!

**Pull request merged**